### PR TITLE
#23 feature view voting add edit

### LINKED ISSUE #23: View: Voting Dialog
# Description
As user, I want to be ablte to add/edit a voting.

# Acceptance criteria
- Todo
- as ui example

# Possible way to test
Todo by developer
1. picture in specific size
2. adding participants
3. all buttons

### PR DESCRIPTION
**Pull request by Islam-Hemida**
# 💻 What
Added Add/Edit Voting page.
Added some test data.
Filled in voting list.
Linked list items to voting page.

# 📘 How
click on Neues Voting button to open the dialog.then fullfill
 the Teilnehmer field and press Hinzufügen.

# 👀 See

addEditEvent

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -26,6 +26,7 @@
     "@angular/platform-browser": "^15.2.0",
     "@angular/platform-browser-dynamic": "^15.2.0",
     "@angular/router": "^15.2.0",
+    "all": "^0.0.0",
```
Comment (bognern) [inline package.json]: What is this package for? Was this added on purpose?

File: src/app/voting/add-edit/add-edit.component.css
Comment (bognern) [inline src/app/voting/add-edit/add-edit.component.css:1]: Since this file contains the same CSS rules as `event-dialog.component.css`, I think it would be better to reference that file in this component's `styleUrls` to avoid duplicate code. Alternatively, you could also create a separate file, put it in a `shared` folder and reference it in both components. Of course, rules that only apply to this component should stay here.

File: src/app/voting/add-edit/add-edit.component.html
Comment (bognern) [inline src/app/voting/add-edit/add-edit.component.html:1]: I think this component could also reuse the `event-dialog` component (see `category-dialog`) since the UI is largely the same. I'm not sure if it's worth the effort it would take to modify it though.

File: src/app/voting/voting-element/voting-card/voting-card.component.html [hunk 2]
```diff
@@ -0,0 +1,13 @@
+<div class="card-holder">
+  <mat-card class="card">
+    <p style="margin: unset">{{ name }}</p>
+    <img src="favicon.ico" height="80px" class="cover" />
+    <p>{{ text }}</p>
+    <div class="flex alignment">
+      <button mat-raised-button class="button" (click)="closeCard()">
+        Close
+      </button>
```
Comment (bognern) [inline src/app/voting/voting-element/voting-card/voting-card.component.html]:
```suggestion
      <button mat-raised-button class="btn btn-primary" (click)="closeCard()">
        Close
      </button>
```

Apply global styles.

File: src/app/voting/voting-element/voting-card/voting-card.component.css [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+.alignment {
+  justify-content: space-between;
+  align-items: baseline;
+}
+
+.button {
+  margin-left: 15px;
+  border-radius: 20px !important;
+  background-color: #fdd516 !important;
+}
```
Comment (bognern) [inline src/app/voting/voting-element/voting-card/voting-card.component.css]:
```suggestion
```

Use global styles instead.

File: src/app/voting/new-voting-button/new-voting-button.component.css [hunk 4]
```diff
@@ -0,0 +1,5 @@
+button {
+  margin-left: 15px;
+  border-radius: 20px !important;
+  background-color: #fdd516 !important;
+}
```
Comment (bognern) [inline src/app/voting/new-voting-button/new-voting-button.component.css]:
```suggestion

```
Use global styles instead.

File: src/app/voting/new-voting-button/new-voting-button.component.html [hunk 5]
```diff
@@ -0,0 +1 @@
+<button (click)="onVtnClick()" mat-raised-button ><mat-icon>add</mat-icon> Neues Voting </button>
```
Comment (bognern) [inline src/app/voting/new-voting-button/new-voting-button.component.html]:
```suggestion
<button (click)="onVtnClick()" class="btn btn-primary" mat-raised-button ><mat-icon>add</mat-icon> Neues Voting </button>
```
Is it necessary to put this button in a separate component?

File: src/app/voting/add-edit/add-edit.component.css [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+  align-items: center;
+  justify-content: center;
+}
+
+.left-align {
+  display: flex;
+  flex-direction: column;
+  align-items: start;
+  justify-content: start;
+}
```
Comment (bognern) [inline src/app/voting/add-edit/add-edit.component.css]:
```suggestion

.box-with-participants {
  background-color: gray;
  width: 200px;
  height: 200px;
}

```

Remove duplicate rules.

File: src/app/voting/add-edit/add-edit.component.ts [hunk 7]
```diff
@@ -0,0 +1,50 @@
+import { Component } from '@angular/core';
+import { FormControl, FormGroup, Validators } from '@angular/forms';
+
+@Component({
+  selector: 'app-add-edit',
+  templateUrl: './add-edit.component.html',
+  styleUrls: ['./add-edit.component.css'],
```
Comment (bognern) [inline src/app/voting/add-edit/add-edit.component.ts]:
```suggestion
  styleUrls: ['./add-edit.component.css', '../../program/shared/event-dialog/event-dialog.component.css'],
```
Reference CSS files.

File: src/app/navigation/navigation.component.html [hunk 8]
```diff
@@ -3,12 +3,12 @@
...
-    </button>
-    <button mat-button routerLink="/voting" routerLinkActive="active-link">
-      Voting
-    </button>
+    <a routerLink="/program" class="button-text-color">
+      <button mat-button>Program</button>
+    </a>
+    <a routerLink="/voting" class="button-text-color">
+      <button mat-button>Voting</button>
+    </a>
```
Comment (bognern) [inline src/app/navigation/navigation.component.html]:
```suggestion
    <button mat-button routerLink="/program" routerLinkActive="active-link">
      Program
    </button>
    <button mat-button routerLink="/voting" routerLinkActive="active-link">
      Voting
    </button>
```

I think these changes should be reverted since the navbar works fine on `develop`. Here, the active page is not highlighted.

File: src/app/voting/voting-element/voting-element.component.html [hunk 9]
```diff
@@ -0,0 +1,30 @@
+<br />
```
Comment (bognern) [inline src/app/voting/voting-element/voting-element.component.html]:
```suggestion
```

It would be better to use `margin` or `padding` to create space between elements.

File: src/app/app-routing.module.ts [hunk 10]
```diff
@@ -25,6 +30,7 @@ const routes: Routes = [
     path: 'program/edit/:id',
     component: EditEventComponent,
   },
+  { path: 'voting', component: AddEditComponent },
```
Comment (bognern) [inline src/app/app-routing.module.ts]:
```suggestion

```
Why did you add the Voting Dialog component to the `/voting` route? There should only be one component for each path.
Also, I think it'd be better to rename `AddEditComponent` so it contains the word `voting`. That would make it clearer which page it belongs to.

File: src/app/voting/add-edit/add-edit.component.html
Comment (Islam-Hemida) [reply src/app/voting/add-edit/add-edit.component.html:1]: No i dont think so, its not at all the same like event-dialog! In Future the logic will be not the same.

File: package.json (hunk 1 above)
Comment (Islam-Hemida) [reply package.json]: deleted

File: src/app/app-routing.module.ts (hunk 10 above)
Comment (Islam-Hemida) [reply src/app/app-routing.module.ts]: renamed as VotingAddEdit!!!

File: src/app/voting/new-voting-button/new-voting-button.component.html (hunk 5 above)
Comment (Islam-Hemida) [reply src/app/voting/new-voting-button/new-voting-button.component.html]: @Gerw4ld

File: src/app/voting/add-edit/add-edit.component.css (hunk 6 above)
Comment (Islam-Hemida) [reply src/app/voting/add-edit/add-edit.component.css]: can you help with the styling issues if you have time?

Comment (Islam-Hemida) [reply src/app/voting/add-edit/add-edit.component.css]: @bognern

**Review by bognern — APPROVED** [review]
Thank you for resolving all issues 👍
